### PR TITLE
VGG19 as Functional Subclass Model and Closures Removed

### LINKED ISSUE #1429: Make VGG a Functional subclass model & remove closures
This should follow the example of #1412 and #1401, including

- Docstring updates
- Colab demonstrating that weight loading works (if pre-trained weights exist) and numerics are identical

### PR DESCRIPTION
**Pull request by srikesh-07**
# What does this PR do?

Fixes #1429 

This [Colab](https://colab.research.google.com/drive/1b-ohYAceuHn9JonR3loHd0-tcE3QM9Vk?usp=sharing) helps to reproduce the same results from existing and new VGG19.

## Who can review?
@ianstenbit  @LukeWood @tanzhenyu

### CODE REVIEW

File: keras_cv/models/vgg19.py [hunk 1]
```diff
@@ -25,39 +25,52 @@
 
 from keras_cv.models import utils
 
+# Copyright 2022 The KerasCV Authors. All Rights Reserved.
```
Comment (ianstenbit) [inline keras_cv/models/vgg19.py:28]: Looks like we accidentally ended up with a second copy of the copyright notice here

Comment (ianstenbit) [reply keras_cv/models/vgg19.py:28]: (and top-level docstring + imports)

File: keras_cv/models/vgg19.py [hunk 2]
```diff
@@ -25,39 +25,52 @@
...
+from keras_cv.models.vgg16 import apply_vgg_block
+
+
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class VGG19(keras.Model):
+    """Instantiates the VGG19 architecture.
     Reference:
     - [Very Deep Convolutional Networks for Large-Scale Image Recognition](https://arxiv.org/abs/1409.1556) (ICLR 2015)
-
     This function returns a Keras VGG19 model.
```
Comment (ianstenbit) [inline keras_cv/models/vgg19.py:61]: nit: this is no lunger a function, but is now a class. (So the "instantiates" comment above is also not quite correct)

File: keras_cv/models/vgg19.py [hunk 3]
```diff
@@ -68,112 +81,145 @@ def VGG19(
...
     """
-    if weights and not tf.io.gfile.exists(weights):
-        raise ValueError(
-            "The `weights` argument should be either `None` or the path to the "
-            "weights file to be loaded. Weights file not found at location: {weights}"
+
+    def __init__(
+        self,
+        input_tensor=None,
+        include_rescaling=True,
```
Comment (ianstenbit) [inline keras_cv/models/vgg19.py:97]: Let's keep these two `include_` arguments as the first two, and not include defaults.

**Review by ianstenbit — CHANGES_REQUESTED** [review]
Thank you!

File: keras_cv/models/vgg19.py (hunk 2 above)
Comment (srikesh-07) [reply keras_cv/models/vgg19.py]: Fixed..

File: keras_cv/models/vgg19.py (hunk 3 above)
Comment (srikesh-07) [reply keras_cv/models/vgg19.py:68]: Fixed, Thank you for guiding through the minor changes. Kindly please review and let me know if any changes to be made.

**Review by srikesh-07 — COMMENTED** [review]
Fixed, Thank you for guiding through the minor changes. Kindly please review and let me know if any changes to be made.

File: keras_cv/models/vgg19.py [hunk 4]
```diff
@@ -14,50 +14,35 @@
...
-
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class VGG19(keras.Model):
+    """
     Reference:
     - [Very Deep Convolutional Networks for Large-Scale Image Recognition](https://arxiv.org/abs/1409.1556) (ICLR 2015)
-
-    This function returns a Keras VGG19 model.
-
+    This class returns a Keras VGG19 model.
```
Comment (ianstenbit) [inline keras_cv/models/vgg19.py:33]: How about "represents" instead of "returns"

File: keras_cv/models/vgg19.py (hunk 3 above)
Comment (ianstenbit) [reply keras_cv/models/vgg19.py:68]: Please remove the defaults for `include_rescaling` and `include_top`

e.g.

```
self,
include_rescaling,
include_top,
classes=None,
...
```

**Review by ianstenbit — CHANGES_REQUESTED** [review]
thanks for the quick edits! Almost ready to merge

File: keras_cv/models/vgg19.py (hunk 4 above)
Comment (srikesh-07) [reply keras_cv/models/vgg19.py]: Fixed.

**Comment by srikesh-07** [comment]
I have also done the changes on VGG19 as well as on VGG16. Kindly please let me know if any change should be made, Thank you.

**Review by ianstenbit — APPROVED** [review]
Looks good -- thank you!

**Pull request merged**